src/main/java/org/apache/commons/math4/stat/descriptive/WeightedEvaluation.java

```diff
@@ -51,7 +56,7 @@
      * do not match, indices are invalid, weights contain NaN, negative or
      * infinite values, or weights does not include at least on positive value
      */
-    double evaluate(double[] values, double[] weights, int begin, int length)
+    double evaluate(double[] values, double[] weights, @IndexFor({"#1", "#2"}) int begin, @NonNegative @LTLengthOf(value = {"#1", "#2"}, offset = {"#3 - 1", "#3 - 1"}) int length)
```

Comment:
Why was the previous version of this method required to have SameLen arrays, but this one is not? Their documentation still says that the lengths must match. Perhaps it is incorrect, since the code would not typecheck otherwise, yes?